src/api/Annotations.js

```diff
@@ -74,24 +82,39 @@ export default class Annotations extends MarkerBasedApi {
         fileId: string,
         annotationId: string,
         permissions: AnnotationPermission,
-        message: string,
+        payload: { message?: string, status?: AnnotationStatus },
         successCallback: (annotation: Annotation) => void,
         errorCallback: (e: ElementsXhrError, code: string) => void,
     ): void {
         this.errorCode = ERROR_CODE_EDIT_ANNOTATION;
+        const { message, status } = payload;
 
-        try {
-            this.checkApiCallValidity(PERMISSION_CAN_EDIT, permissions, fileId);
-        } catch (e) {
-            errorCallback(e, this.errorCode);
-            return;
+        if (message) {
+            try {
+                this.checkApiCallValidity(PERMISSION_CAN_EDIT, permissions, fileId);
+            } catch (e) {
+                errorCallback(e, this.errorCode);
+                return;
+            }
         }
 
-        const requestData = { data: { description: { message } } };
+        if (status) {
+            try {
+                this.checkApiCallValidity(PERMISSION_CAN_RESOLVE, permissions, fileId);
+            } catch (e) {
+                errorCallback(e, this.errorCode);
+                return;
+            }
+        }
 
         this.put({
             id: fileId,
-            data: requestData,
+            data: {
+                data: {
+                    description: { message },
```

Comment:
I didn't realize that `description` accepted an object here. We should probably do `description: message ? { message } : undefined`, instead.